src/client/app/effects/github.ts

```diff
@@ -109,9 +110,15 @@ export class GitHubEffects {
             }
 
             return this._github.createOrUpdateGist(
-                description, files, gistId, type === GitHub.GitHubActionTypes.SHARE_PUBLIC_GIST
+                    description, files, gistId, type === GitHub.GitHubActionTypes.SHARE_PUBLIC_GIST
             )
-            .map((gist: IGist) => ({ gist: gist, snippetId: id }));
+            .map((gist: IGist) => ({ gist: gist, snippetId: id }))
+            .catch(exception => {
+                if (exception.status === 404) {
+                    throw { id: id, gistOwnerId: UNKNOWN_GIST_OWNER_ID};
```

Comment:
I added it as a new `Error`, with its message field containing the JSON object with appropriate parameters. The `exception` in the `catch` blocks in this function is normally of type `Response` so throwing something of type `Error` should be enough to distinguish it I believe.